diff -Nur a/linux/net/ipsec/radij.c openswan-2.6.41/linux/net/ipsec/radij.c
--- a/linux/net/ipsec/radij.c	2015-09-16 16:51:51.020603162 -0700
+++ openswan-2.6.41/linux/net/ipsec/radij.c	2015-09-16 16:54:14.948606787 -0700
@@ -773,9 +773,9 @@
 	 * the successor node in advance.
 	 */
 	/* First time through node, go left */
-	while (rn->rj_b >= 0)
+	while (rn && rn->rj_b >= 0)
 		rn = rn->rj_l;
-	for (;;) {
+	for (;rn;) {
 		if(debug_radij) {
 			printk("klips_debug:rj_walktree: "
 			       "for: rn=0p%p rj_b=%d rj_flags=%x",
@@ -792,13 +792,13 @@
 		}
 		base = rn;
 		/* If at right child go back up, otherwise, go right */
-		while (rn->rj_p->rj_r == rn && (rn->rj_flags & RJF_ROOT) == 0)
+		while (rn && rn->rj_p->rj_r == rn && (rn->rj_flags & RJF_ROOT) == 0)
 			rn = rn->rj_p;
 		/* Find the next *leaf* since next node might vanish, too */
-		for (rn = rn->rj_p->rj_r; rn->rj_b >= 0;)
+		for (rn = rn->rj_p->rj_r;rn && rn->rj_b >= 0;)
 			rn = rn->rj_l;
 		next = rn;
-		if(debug_radij) {
+		if(rn && debug_radij) {
 			printk("klips_debug:rj_walktree: "
 			       "processing leaves, rn=0p%p rj_b=%d rj_flags=%x",
 			       rn,
@@ -834,9 +834,10 @@
 				return (-error);
 		}
 		rn = next;
-		if (rn->rj_flags & RJF_ROOT)
+		if (rn && rn->rj_flags & RJF_ROOT)
 			return (0);
 	}
+	return (0);
 	/* NOTREACHED */
 }
 
